### ISSUE TITLE
Enquiry about Training CLIP-based Model

### ISSUE THREAD

**Issue by ghost** [MCG-NJU/Dynamic-MDETR#1]
Hi, thanks for your work. Could you please provide the training script for the clip-base models? It seems there are only scrpits for training the ResNet-based models. Thank you!

**Comment by ShiFengyuan1999** [comment]
Hi @Yorick-creator, you can try the following training script for clip-base model training.

### refcocog-g
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset gref --max_query_len 80 --output_dir outputs/refcocog_gsplit_clip  --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 30 --epochs 40 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096`
### refcocog-u
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset gref_umd --max_query_len 80 --output_dir outputs/refcocog_usplit_clip  --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 30 --epochs 40 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096`
### refcoco
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset unc --max_query_len 40 --output_dir outputs/refcoco_clip  --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 30 --epochs 40 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096`
### refcoco plus
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset unc+ --max_query_len 40 --output_dir outputs/refcoco_plus_clip --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 30 --epochs 40 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096`
### referit
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset referit --max_query_len 40 --output_dir outputs/referit_clip  --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 45 --epochs 60 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096 --dropout 0.0 --vl_dropout 0.0 --weight_decay 0.0`
### flickr
`python -m torch.distributed.launch --nproc_per_node=8 --use_env train.py --model_typy CLIP --batch_size 8 --aug_crop --aug_scale --aug_translate --detr_enc_num 6 --dataset flickr --max_query_len 40 --output_dir outputs/flickr_clip  --stages 3 --vl_fusion_enc_layers 3 --uniform_learnable True --in_points 160 --lr 1e-4 --different_transformer True --lr_drop 15 --epochs 20 --vl_dec_layers 1 --vl_enc_layers 1 --clip_max_norm 1.0 --imsize 640 --vl_hidden_dim 512 --vl_dim_feedforward 4096`

**Comment by ghost** [comment]
Thank you very much!
